lua/bufferline/render.lua

```diff
@@ -131,14 +134,16 @@ local function render()
       icon =
         (letter ~= nil and letter or ' ') ..
         (has_icons and ' ' or '')
-    elseif has_icons then
+    else
       if has_numbers then
         local number_text = tostring(i)
         iconPrefix = ''
-        icon = number_text .. (#number_text > 1 and '' or ' ')
-      else
+        icon = icon .. number_text .. ' '
+      end
+
+      if has_icons then
         local iconChar, iconHl = get_icon(buffer_name, vim.fn.getbufvar(buffer_number, '&filetype'))
-        iconPrefix = status == 'Inactive' and hl('BufferInactive') or hl(iconHl or ('Buffer' .. status))
+        iconPrefix = icon .. (is_inactive and hl('BufferInactive') or hl(iconHl or ('Buffer' .. status)))
```

Comment:
To enable "sharing" the space with the `g:bufferline.icons.numbers` setting, I include the `numbers` output in the `iconPrefix` so that its highlight group won't get clobbered by `hl('BufferInactive')`.